blog/_posts/2021-07-19-scala-3.0.2RC1-is-here.md

```diff
@@ -0,0 +1,224 @@
+---
+layout: blog-detail
+post-type: blog
+by: Michał Pałka, VirtusLab
+title: Scala 3.0.1 and 3.0.2-RC1 are here!
+---
+
+Greetings from the Scala 3 team! We are glad to announce that Scala 3.0.1 and 3.0.2-RC1 are now officially out.
+As no critical bugs have been found in the previously released Scala 3.0.1-RC2, it has been promoted to 3.0.1, which is the first stable release after 3.0.0. Scala 3.0.2-RC1, in turn, incorporates new language improvements and bug fixes.
+
+# Improved insertion of semicolons in logical conditions
+
+Scala 3's indentation based syntax is aimed at making your code more concise and readable. As it gets broader adoption, we consistently improve its specification to eliminate corner cases which might lead to ambiguities or counterintuitive behaviours.
+
+Thanks to [#12801](https://github.com/lampepfl/dotty/pull/12801) it is now allowed for a logical expression in an `if` statement or expression to continue in the following line if it starts in the same line as the `if` keyword, e.g.
+
+```scala
+if foo
+  (bar)
+then //...
+```
+
+can now be used instead of
+
+```scala
+if foo(bar)
+then //...
+```
+
+If yor intention is to have a block of code evaluating into a sigle condition you should add a new line and indentation directly after `if` , e.g.
+
+```scala
+if
+  val cond = foo(bar)
+  cond
+then //...
+```
+
+so code like below would NOT be valid
+
+```scala
+if val cond = foo(bar)
+  cond
+then //...
+```
+
+# Towards better null safety in the type system
+
+The compiler option `-Yexplicit-nulls` modifies Scala's standard type hierarchy to allow easier tracing of nullable values by performing strict checks directly on the level of the type system rather than just relying of conventions (e.g. this prevents you from writing code like `val foo: Option[String] = Some(null)`, which would be otherwise valid Scala although very likely to cause a `NullPointerException` at some further point).
```

Comment:
```suggestion
The compiler option `-Yexplicit-nulls` modifies Scala's standard type hierarchy to allow easier tracing of nullable values by performing strict checks directly on the level of the type system rather than just relying on conventions (e.g. this prevents you from writing code like `val foo: Option[String] = Some(null)`, which would be otherwise valid Scala although very likely to cause a `NullPointerException` at some further point).
```